extractor/src/main/java/org/schabi/newpipe/extractor/linkhandler/ListLinkHandlerFactory.java

```diff
@@ -1,24 +1,29 @@
 package org.schabi.newpipe.extractor.linkhandler;
 
 import org.schabi.newpipe.extractor.exceptions.ParsingException;
+import org.schabi.newpipe.extractor.search.filter.FilterItem;
 import org.schabi.newpipe.extractor.utils.Utils;
 
-import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
 public abstract class ListLinkHandlerFactory extends LinkHandlerFactory {
 
     ///////////////////////////////////
     // To Override
     ///////////////////////////////////
 
-    public abstract String getUrl(String id, List<String> contentFilter, String sortFilter)
+    public abstract String getUrl(String id,
+                                  @Nonnull List<FilterItem> contentFilter,
+                                  @Nullable List<FilterItem> sortFilter)
             throws ParsingException, UnsupportedOperationException;
 
     public String getUrl(final String id,
-                         final List<String> contentFilter,
-                         final String sortFilter,
+                         final List<FilterItem> contentFilter,
+                         final List<FilterItem> sortFilter,
```

Comment:
Some other `@Nonnull` and `@Nullable` annotations are missing in this file (and all derivate classes)